packages/@webex/plugin-cc/src/AgentProfileService/AgentProfileService.ts

```diff
@@ -0,0 +1,146 @@
+    } catch (error) {
+      this.webex.logger.error(`Error while calling retrieveDesktopProfileById API, ${error}`);
+
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Method to get List of Teams.
+   * @param {string} orgId The ID of an Organization.
+   * @param {number} page (Optional) Index of the page of results to be fetched.
+   * @param {number} pageSize (Optional) Number of items to be displayed on a page.
+   * @param {Array<String>} filter (Optional) Filter which can be applied to the elements to be fetched.
+   * @param {Array<String>} attributes (Optional) Specify the attributes to be returned. Supported attributes are id, name, active and workTypeCode.
+   * @returns {Promise<ListTeamsResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getListOfTeams(
+    orgId: string,
+    page?: number,
+    pageSize?: number,
+    filter?: string[],
+    attributes?: string[]
+  ): Promise<ListTeamsResponse> {
+    try {
+      if (!orgId) Promise.reject(new Error('orgId is undefined'));
+      const URL = `${this.wccAPIURL}organization/${orgId}/team?page=${page}&pageSize=${pageSize}&filter=id=in=${filter}&attributes=${attributes}`;
+
+      return Promise.resolve(this.makeGETRequest(URL));
+    } catch (error) {
+      this.webex.logger.error(`Error while calling getListOfTeams API, ${error}`);
+
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Method to get List of AuxCodes.
+   * @param {string} orgId The ID of an Organization.
+   * @param {number} page (Optional) Index of the page of results to be fetched.
+   * @param {number} pageSize (Optional) Number of items to be displayed on a page.
+   * @param {Array<String>} filter (Optional) Filter which can be applied to the elements to be fetched.
+   * @param {Array<String>} attributes (Optional) Specify the attributes to be returned. Supported attributes are id, name, active and workTypeCode.
+   * @returns {Promise<ListAuxCodesResponse>} A promise that eventually resolves to an API response.
+   */
+
+  public async getListOfAuxCodes(
+    orgId: string,
+    page?: number,
+    pageSize?: number,
+    filter?: string[],
+    attributes?: string[]
+  ): Promise<ListAuxCodesResponse> {
+    try {
+      if (!orgId) Promise.reject(new Error('orgId is undefined'));
+      const URL = `${this.wccAPIURL}organization/${orgId}/v2/auxiliary-code?page=${page}&pageSize=${pageSize}&filter=id=in=${filter}&attributes=${attributes}`;
+
+      return Promise.resolve(this.makeGETRequest(URL));
+    } catch (error) {
+      this.webex.logger.error(`Error while calling getListOfAuxCodes API, ${error}`);
+
+      return Promise.reject(error);
+    }
+  }
+
+  /**
+   * Common method to make GET Requests.
+   * @param {string} URL The URL of the Request.
+   * @returns {Promise<any>} A promise that eventually resolves to an individual API response.
+   */
+
+  public async makeGETRequest(URL: string): Promise<any> {
```

Comment:
Does this function add any utility to the class? Does it or will it do any extra operations in the future?
As of now, doesn't this look like a wrapper?

Reply:
In order to comply with the "DO-NOT-REPEAT-YOURSELF" strategic I made this common method that will be used by the above 4 CC APIs. These all 4 methods makes a GET request and the only change is in the URL. So, I introduced this method. We can create a new file request.ts inside the src folder and that file can contain this method as in future we will be working on more CC feature so we might need this method to make GET request.

Reply:
This makes sense if we have a little more processing in the function. However, as of now it only seems like we'll be doing a webex.request everywhere right?